backend/src/app.py

```diff
@@ -19,10 +21,22 @@
 )
 
 
+@app.errorhandler(UnauthorizedException)
+def handle_unauthorized_exception(_):
+    resp = make_response(
+        "Spotify access token invalid or missing. Please re-authenticate.", 401
+    )
+    resp.delete_cookie("spotify_access_token")
+    resp.delete_cookie("user_id")
+    return resp
+
+
 @app.route("/")
 def index():
     user_id = request.cookies.get("user_id")
     access_token = request.cookies.get("spotify_access_token")
+    if access_token is None:
+        raise UnauthorizedException
```

Comment:
Nothing wrong with this, but it strikes me as some logic that:
* Is going to end up duplicated
* Is going to end up missed somewhere

Would this maybe be a better check to put directly on the BearerAuth class, which we trust all Spotify requests will use?

Reply:
Ah, that's a much cleaner way of doing it! Good shout!